test/opt/pass_utils.cpp

```diff
@@ -43,18 +43,35 @@ const char* kDebugOpcodes[] = {
     // clang-format on
 };
 
-// Returns true if the given string contains any debug opcode substring.
+}  // anonymous namespace
+
+namespace spvtools {
+
+bool ReplaceSubstringInPlace(std::string* process_str, const std::string find_str,
+                          const std::string replace_str) {
+  if (process_str->empty()) {
+    return false;
+  }
+  bool replaced = false;
+  size_t pos = 0;
+  while(pos != std::string::npos) {
```

Comment:
Comments should say that this algorithm is quadratic.  If someone copies this into another context, then they probably want a better algorithm .